packages/engine/paima-sm/src/types.ts

```diff
@@ -262,6 +288,20 @@ export type ChainDataExtensionCardanoDelegation = ChainDataExtensionBase &
     cdeType: ChainDataExtensionType.CardanoPool;
   };
 
+export const ChainDataExtensionCardanoProjectedNFTConfig = Type.Object({
+  type: Type.Literal(CdeEntryTypeName.CardanoProjectedNFT),
+  contract: Type.String(),
+  scheduledPrefix: Type.String(),
```

Comment:
Probably this should be option since most games will probably not need to update their state machine based on somebody locking or queuing to unlock

Probably what is more important is the `getCardanoAddressProjectedNfts` function you wrote

Reply:
I probably don't fully understand the role of `scheduledPrefix`, could you explain pls?

Reply:
`scheduledPrefix` is the event that get scheduled to be sent over to the state machine for the game

For example, you can see entrypoint for the state machine of a game [here](https://github.com/PaimaStudios/paima-game-templates/blob/main/rock-paper-scissors/state-transition/src/stf/v1/index.ts#L28) use a switch-case on the prefix of the message it receives from Paima.